inst/include/Rcpp/exceptions.h

```diff
@@ -89,6 +89,48 @@ namespace Rcpp {
             file_io_error("file already exists", file) {}	// #nocov end
     };
 
+    #define RCPP_ADVANCED_EXCEPTION_CLASS(__CLASS__, __WHAT__)                                                                                                                                                     \
```

Comment:
It would be very nice to have Vardiac template support... Mmm. 

Would something along the lines of:

```
#define RCPP_CPP11_EXCEPTION_CLASS(__CLASS__, __WHAT__)     
     class __CLASS__ : public std::exception{                                       \
     public:                                                                        \
         __CLASS__( const std::string& message ) throw() : message( __WHAT__ ){} ;  \
        template <typename... Args>                                                 \
         __CLASS__( const char* fmt, Args&&... args ) throw() :                     \
         message(  tfm::format(fmt, std::forward<Args>(args)...) ){} ;              \
         virtual ~__CLASS__() throw(){} ;                                           \
         virtual const char* what() const throw() { return message.c_str() ; }      \
     private:                                                                       \
         std::string message ;                                                      \
     } ;
```

Be acceptable? 

**Note:** The defines in [`/inst/include/Rcpp/utils/tinyformat.h`](https://github.com/RcppCore/Rcpp/blob/6f81b4684481dbd9bb554dd95e66725fc3b63a8c/inst/include/Rcpp/utils/tinyformat.h#L131-L136) will also need to receive the `if __cplusplus >= 201103L` check.